transcom/mymove
# Int b 18171 hal prime actions cause sit to be calculated

## [B-18171 Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A871863)

## Summary
PPM shipment #6A9DE, Domestic destination SIT delivery
Status: SUBMITTED

> Requested SIT delivery date: 17 Feb 2024
> Customer contacted: 27 Feb 2024

<img width="1223" alt="image" src="https://github.com/transcom/mymove/assets/148263228/345bca7a-20c1-485c-b4e1-a69442fb86d4">


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

### How to test
- Setup a NAVY service member with a PPM:
  - Set the origin duty station to NS Mayport, FL 32228.
  - Fill out the rest of the items.

- Create a multi-user on the GBLOC for the service member:
  - Obtain the locator by running the following query in DBeaver:
    ```sql
    SELECT orders.gbloc, moves.locator 
    FROM mto_service_items
    JOIN moves ON moves.id = mto_service_items.move_id 
    JOIN orders ON orders.id = moves.orders_id 
    WHERE mto_service_items.id ='THE_GUID_GOES_HERE';
    ```
  - Pick the appropriate GBLOC from the results.

- Log in as a TOO (Transportation Operations Officer):
  - Fill out the actionable items such as orders details.
  - Check PPM and submit the move with the counseling check field checked to enable the button.

- Log in as a Prime:
  - Add a destination SIT (Storage-in-Transit) service item by filling in the details and getting the item in.

- Go back to the TOO user:
  - Reject the service items.

- Log in as a Prime:
  - edit the DDDSIT - Domestic destination SIT delivery and  Change the Departure Date, Requested Delivery and Customer Contacted dates and hit save.
  - Go to the prime update shipment screen (by clicking the update shipment button, for the ppm)
  - Select SIT Expected, and set the location to 'destination' or 'origin' and set the 'SIT Estimated Weight (lbs)' and dates. 
  - Change the Estimated Weight (lbs) under the 'Weights' label
  - Click 'save'
- Go back to the TOO user:
  - Inspect the audit log for the line items:

You should see
<img width="1223" alt="image" src="https://github.com/transcom/mymove/assets/148263228/345bca7a-20c1-485c-b4e1-a69442fb86d4">
These are the items you should see, at least.
- Sit expected
- SIT Location
- SIT estimated storage weight
- SIT Estimated Cost:
- PPM Estimated Incentive:

<img width="1223" alt="Pasted Graphic 9" src="https://github.com/transcom/mymove/assets/148263228/ba344596-b1db-40fa-a5a5-21d65b3bb49b">
These are the items you should see when the service item was created.
- SIT Estimated Entry Date
- SIT Estimated Departure Date

This meets the acceptance criteria. If you have trouble accessing the closeout, form on dates and locations for the ppm, it’s because the airfare and army are conditionally checked for it, and not the other branches.

## Verification Steps for the Author
